ios/brave-ios/Sources/Brave/Frontend/Settings/Display/NTPTableViewController.swift

```diff
@@ -84,21 +90,41 @@ class NTPTableViewController: TableViewController {
     dataSource.sections = [imageSection, widgetSection]
   }
 
-  private func selectedItem() -> BackgroundImageType {
+  private func selectedItem() -> BackgroundType {
     if let referrer = Preferences.NewTabPage.selectedCustomTheme.value {
       return .superReferrer(referrer)
     }
 
-    return Preferences.NewTabPage.backgroundSponsoredImages.value ? .sponsored : .defaultImages
+    switch Preferences.NewTabPage.backgroundMediaType {
+    case .defaultImages:
+      return BackgroundType.defaultImages
+    case .sponsoredImages:
+      return BackgroundType.sponsoredImages
+    case .sponsoredImagesAndVideos:
+      return rewards?.shouldShowSponsoredImagesAndVideos == true
+        ? BackgroundType.sponsoredImagesAndVideos(
+          Strings.NTP.settingsSponsoredImagesAndVideosSelection
+        )
+        : BackgroundType.sponsoredImagesAndVideos(Strings.NTP.settingsSponsoredImagesSelection)
+    }
```

Comment:
A bit confused by this, if `shouldShowSponsoredImagesAndVideos` is false, shouldn't `BackgroundType. sponsoredImagesAndVideos` media type be undefined behaviour?

Reply:
The `shouldShowSponsoredImagesAndVideos` actually controls whether we show `Sponsored Images & Videos` option or not. It is a temporary workaround for showing this option for Japan region only. We will remove it when we are able to run video NTT for other countries.

Under the hood the default value for the `backgroundMediaType` setting is `.sponsoredImagesAndVideos` which works for both image and video backgrounds. This will allow us not to do a migration when `Sponsored Images & Videos` option will be available for other countries.

Reply:
I will rename `shouldShowSponsoredImagesAndVideos` function to `shouldShowSponsoredImagesAndVideosSetting` to make it clear. Will add a proper code comment as well.